pkg/binding/crd.go

```diff
@@ -35,19 +37,43 @@ type APIResource struct {
 }
 
 // Handle CRDs should account for CRDs being added or deleted to start/stop new informers as needed
-func (c *Controller) handleCRD(_ runtime.Object) error {
-	toStartList, toStopList, err := c.checkAPIResourcesForUpdates()
-	if err != nil {
-		return err
+func (c *Controller) handleCRD(obj runtime.Object) error {
+	uObj := obj.(*unstructured.Unstructured)
+	var crdObj *apiextensionsv1.CustomResourceDefinition
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(uObj.UnstructuredContent(), &crdObj); err != nil {
+		return fmt.Errorf("failed to convert Unstructured to CRD: %w", err)
+	}
+
+	toStart := APIResource{
+		groupVersion: schema.GroupVersion{
+			Group:   crdObj.Spec.Group,
+			Version: crdObj.Spec.Versions[0].Name,
```

Comment:
I wonder if it is safe to look at `[0]` and no other. I thought that `v1` and `v2` have distinct populations of objects. Also, I looked at https://github.com/kubernetes/apiextensions-apiserver/blob/v0.28.2/pkg/apis/apiextensions/v1/types.go ; could `[0]` have `Served==false`?

Reply:
Added in https://github.com/kubestellar/kubestellar/pull/1828/commits/c1aa6c0bc2656c2209108008cecb00243ed7dd73 the functionality to manage the watches of multiple versions of a CRD.